developers_chamber/scripts/qa.py

```diff
@@ -0,0 +1,250 @@
+import re
+import subprocess
+
+from developers_chamber.scripts import cli
+from developers_chamber.utils import print_bold, print_error, print_heading, print_success
+from git import Repo
+
+
+class QAError(Exception):
+    def __init__(self, msg, output):
+        super().__init__(msg)
+        self.output = output.strip()
+
+
+class QACheck:
+    """
+    Base class for a quality assurance check.
+
+    Arguments:
+        name: Name of the check.
+    """
+    name = None
+
+    def _get_repo(self):
+        """
+        Returns the repo object.
+        """
+        return Repo('.')
+
+    def _get_default_branch(self):
+        """
+        Returns default branch of the repo.
+        """
+        return self._run_command("git symbolic-ref refs/remotes/origin/HEAD | sed 's@^refs/remotes/origin/@@'")
+
+    def _get_diffs(self):
+        """
+        Returns Git diffs against default branch.
+        """
+        repo = self._get_repo()
+        target_branch = getattr(repo.remotes.origin.refs, self._get_default_branch())
```

Comment:
I need to look onto remote branch since Circle CI hard resets next to the feature branch's commit - that's non-sense.